server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/rewrite/selector/RewriteCompactionFileSelector.java

```diff
@@ -167,6 +167,12 @@ void select(boolean useTightBound) throws IOException {
     while (unseqIndex < resource.getUnseqFiles().size() && timeConsumption < timeLimit) {
       // select next unseq files
       TsFileResource unseqFile = resource.getUnseqFiles().get(unseqIndex);
+      if (!unseqFile.getTsFile().exists()
+          || unseqFile.isCompacting()
+          || !unseqFile.isClosed()
+          || unseqFile.isCompactionCandidate()) {
+        break;
+      }
```

Comment:
Only need to judge whether the file exists, because the others are filtered out when the `CompactResource` is constructed earlier.

Reply:
reoslve